astropy/visualization/wcsaxes/core.py

```diff
@@ -641,6 +641,94 @@ def grid(self, b=None, axis='both', *, which='major', **kwargs):
         else:
             raise ValueError('axis should be one of x/y/both')
 
+    def tick_params(self, axis='both', **kwargs):
+        """
+        Method to set the tick and tick label parameters in the same way as the
+        :meth:`~matplotlib.axes.Axes.tick_params` method in Matplotlib.
+
+        This is provided for convenience, but the recommended API is to use
+        :meth:`~astropy.visualization.wcsaxes.CoordinateHelper.set_ticks`,
+        :meth:`~astropy.visualization.wcsaxes.CoordinateHelper.set_ticklabel`,
+        :meth:`~astropy.visualization.wcsaxes.CoordinateHelper.set_ticks_position`,
+        :meth:`~astropy.visualization.wcsaxes.CoordinateHelper.set_ticklabel_position`,
+        and :meth:`~astropy.visualization.wcsaxes.CoordinateHelper.grid`.
+
+        Parameters
+        ----------
+        axis : int or str, optional
+            Which axis to apply the parameters to. This defaults to 'both'
+            but this can also be set to an `int` or `str` that refers to the
+            axis to apply it to, following the valid values that can index
+            ``ax.coords``. Note that ``'x'`` and ``'y``' are also accepted in
+            the case of rectangular axes.
+        which : {'major', 'minor', 'both'}, optional
+            Which ticks to apply the settings to. By default, setting are
+            applied to both major and minor ticks. Note that if ``'minor'`` is
+            specified, only the length of the ticks can be set currently.
+        direction : {'in', 'out'}
```

Comment:
optional (all the parameters below are also optional)

Reply:
Fixed